diff --git a/dist/types/KeySortedContext.d.ts b/dist/types/KeySortedContext.d.ts
index afef799bf340b9b9e1b7b6e159d0679c0f5e14e1..73a338984d082ca85839ccd32cc237629c85114b 100644
--- a/dist/types/KeySortedContext.d.ts
+++ b/dist/types/KeySortedContext.d.ts	
@@ -3,9 +3,9 @@
  * Licensing: MIT
  */
 import type { Selector } from '@tsdotnet/common-interfaces';
-import { Comparable, Comparer, Comparison } from './Comparable';
-import { OrderValue as Order } from './Order';
-import SortContext from './SortContext';
+import { Comparable, Comparer, Comparison } from './Comparable.js';
+import { OrderValue as Order } from './Order.js';
+import SortContext from './SortContext.js';
 export default class KeySortedContext<T, TKey extends Comparable> extends SortContext<T> {
     protected _keySelector: Selector<T, TKey> | null;
     constructor(next: Comparer<T> | null, keySelector: Selector<T, TKey> | null, order?: Order, comparer?: Comparison<T>);
diff --git a/dist/types/OrderBy.d.ts b/dist/types/OrderBy.d.ts
index ee1484d9c6cb097475e57900c1b45739b3afb0fd..dd55598a87040a0adfb74eb1e2c4a0693d40cb05 100644
--- a/dist/types/OrderBy.d.ts
+++ b/dist/types/OrderBy.d.ts	
@@ -4,8 +4,8 @@
  */
 
 import type {Primitive, Selector} from '@tsdotnet/common-interfaces';
-import type {Comparison} from './Comparable';
-import type {OrderValue as Order} from './Order';
+import type {Comparison} from './Comparable.js';
+import type {OrderValue as Order} from './Order.js';
 
 export type OrderByOption = {
 	order?: Order;
diff --git a/dist/types/SortContext.d.ts b/dist/types/SortContext.d.ts
index 83c5094e9a026c711a40ec8ceb8ff74a4109fac8..bbc6c1c361e66a3aec30106854185050ea1134c1 100644
--- a/dist/types/SortContext.d.ts
+++ b/dist/types/SortContext.d.ts	
@@ -2,8 +2,8 @@
  * @author electricessence / https://github.com/electricessence/
  * Licensing: MIT
  */
-import type { Comparer, Comparison } from './Comparable';
-import { OrderOrValue as Order } from './Order';
+import type { Comparer, Comparison } from './Comparable.js';
+import { OrderOrValue as Order } from './Order.js';
 export default class SortContext<T> implements Comparer<T> {
     protected _next: Comparer<T> | null;
     protected _comparer: Comparison<T>;
diff --git a/dist/types/areSequencesEqual.d.ts b/dist/types/areSequencesEqual.d.ts
index ef933ef6273eb1f0e6b4f6eda325373185e7e003..790beda31901a3be6f025c3f486bd00a03f3cbe3 100644
--- a/dist/types/areSequencesEqual.d.ts
+++ b/dist/types/areSequencesEqual.d.ts	
@@ -2,5 +2,5 @@
  * @author electricessence / https://github.com/electricessence/
  * Licensing: MIT
  */
-import type { EqualityComparison } from './Comparable';
+import type { EqualityComparison } from './Comparable.js';
 export default function areSequencesEqual<T>(a: Iterable<T>, b: Iterable<T>, equalityComparer?: EqualityComparison<T>): boolean;
diff --git a/dist/types/compare.d.ts b/dist/types/compare.d.ts
index fb96fddec60bdf78d24ec739bf98fdd212e607c2..0ac0348f4dc5dcc5c54ffc3264883c73578855ee 100644
--- a/dist/types/compare.d.ts
+++ b/dist/types/compare.d.ts	
@@ -3,9 +3,9 @@
  * Licensing: MIT
  */
 import type { Primitive } from '@tsdotnet/common-interfaces';
-import { type ComparableObject } from './Comparable';
-import comparePrimitives from './comparePrimitives';
-import { type CompareResultValue as CompareResult } from './CompareResult';
+import { type ComparableObject } from './Comparable.js';
+import comparePrimitives from './comparePrimitives.js';
+import { type CompareResultValue as CompareResult } from './CompareResult.js';
 declare function compare<T>(a: ComparableObject<T>, b: T): number;
 declare function compare<T>(a: T, b: ComparableObject<T>): number;
 declare function compare<T extends Primitive>(a: T, b: T): CompareResult;
diff --git a/dist/types/comparePrimitives.d.ts b/dist/types/comparePrimitives.d.ts
index 36d612645ef61491fdb75149331242537576f37d..90784237dde5d3c01ac092a9ae0b63446598f009 100644
--- a/dist/types/comparePrimitives.d.ts
+++ b/dist/types/comparePrimitives.d.ts	
@@ -3,7 +3,7 @@
  * Licensing: MIT
  */
 import type { Primitive } from '@tsdotnet/common-interfaces';
-import { CompareResultValue as CompareResult } from './CompareResult';
+import { CompareResultValue as CompareResult } from './CompareResult.js';
 declare function comparePrimitives<T extends Primitive>(a: T, b: T): CompareResult;
 declare namespace comparePrimitives {
     function inverted<T extends Primitive>(a: T, b: T): CompareResult;
diff --git a/dist/types/comparison.d.ts b/dist/types/comparison.d.ts
index ec54bbe193e28f6bac7e7da778571938854fcb04..eaf6362159cd8b95cb6cc3213ac338b5ad43296e 100644
--- a/dist/types/comparison.d.ts
+++ b/dist/types/comparison.d.ts	
@@ -3,9 +3,9 @@
  * Licensing: MIT
  */
 import type { Primitive, Selector } from '@tsdotnet/common-interfaces';
-import type { Comparison } from './Comparable';
-import { OrderValue as Order } from './Order';
-import type { OrderByComparison, OrderByKey, OrderBySelector } from './OrderBy';
+import type { Comparison } from './Comparable.js';
+import { OrderValue as Order } from './Order.js';
+import type { OrderByComparison, OrderByKey, OrderBySelector } from './OrderBy.js';
 export declare function fromSelector<T>(selector: Selector<T, Primitive>, order?: Order): Comparison<T>;
 export declare function fromKey<T extends object>(key: keyof T, order?: Order): Comparison<T>;
 export declare function fromKeys<T extends object>(keys: (keyof T)[]): Comparison<T>;
diff --git a/dist/types/index.d.ts b/dist/types/index.d.ts
index 8295778376db6534d864baf622152eac63a016a8..febfc1eb13e6280df9de2f41edd9967eb8a8b773 100644
--- a/dist/types/index.d.ts
+++ b/dist/types/index.d.ts	
@@ -1,14 +1,14 @@
 import typeUtil from '@tsdotnet/type';
-import areEqual from './areEqual';
-import areEquivalentObjects from './areEquivalent';
-import areEquivalent from './areEquivalent';
-import areSequencesEqual from './areSequencesEqual';
-import type { Comparer, ComparableObject, Comparable, Comparison, EqualityComparison } from './Comparable';
-import * as comparison from './comparison';
-import compare from './compare';
-import CompareResult from './CompareResult';
-import KeySortedContext from './KeySortedContext';
-import Order, { type OrderOrValue } from './Order';
-import SortContext from './SortContext';
-import EqualityOperator from './EqualityOperator';
+import areEqual from './areEqual.js';
+import areEquivalentObjects from './areEquivalent.js';
+import areEquivalent from './areEquivalent.js';
+import areSequencesEqual from './areSequencesEqual.js';
+import type { Comparer, ComparableObject, Comparable, Comparison, EqualityComparison } from './Comparable.js';
+import * as comparison from './comparison.js';
+import compare from './compare.js';
+import CompareResult from './CompareResult.js';
+import KeySortedContext from './KeySortedContext.js';
+import Order, { type OrderOrValue } from './Order.js';
+import SortContext from './SortContext.js';
+import EqualityOperator from './EqualityOperator.js';
 export { areEqual, areEquivalent, areEquivalentObjects, areSequencesEqual, compare, typeUtil as type, type Comparer, type ComparableObject, type Comparable, type Comparison, type EqualityComparison, comparison, CompareResult, EqualityOperator, Order, OrderOrValue, SortContext, KeySortedContext };
