diff --git a/dist/types/LinkQueue.d.ts b/dist/types/LinkQueue.d.ts
index 1135c91efb6cfef497d2137beeff8c7361119759..d2a3b287a340d925a63e57f66f14c4271e618e3e 100644
--- a/dist/types/LinkQueue.d.ts
+++ b/dist/types/LinkQueue.d.ts	
@@ -2,7 +2,7 @@
  * @author electricessence / https://github.com/electricessence/
  * @license MIT
  */
-import QueueBase from './QueueBase';
+import QueueBase from './QueueBase.js';
 export default class LinkQueue<T> extends QueueBase<T> {
     private _root;
     private _tail?;
diff --git a/dist/types/Queue.d.ts b/dist/types/Queue.d.ts
index 94b5b94ff979c2892783e24525f2970e3e18d8c3..86434a54b79ab7071017ad22661bf28d0d767206 100644
--- a/dist/types/Queue.d.ts
+++ b/dist/types/Queue.d.ts	
@@ -1,4 +1,4 @@
-import QueueBase from './QueueBase';
+import QueueBase from './QueueBase.js';
 export default class Queue<T> extends QueueBase<T> {
     protected _array: T[];
     protected _head: number;
diff --git a/dist/types/index.d.ts b/dist/types/index.d.ts
index d63ba7b85780caa9c50037e2b28ccfcc486ea7b5..1a44b34779071bac4e59802eb547e1e824b0b6d1 100644
--- a/dist/types/index.d.ts
+++ b/dist/types/index.d.ts	
@@ -1,5 +1,5 @@
-import Queue from './Queue';
+import Queue from './Queue.js';
 export { Queue };
-export { default as LinkQueue } from './LinkQueue';
-export { default as QueueBase } from './QueueBase';
+export { default as LinkQueue } from './LinkQueue.js';
+export { default as QueueBase } from './QueueBase.js';
 export default Queue;
