diff --git a/dist/types/ArgumentException.d.ts b/dist/types/ArgumentException.d.ts
index f0c7fa407dde7aa8730e30f02cd077026a7d19f0..78e09d1ef6f43fb2c37a20bfa1f212d9d36f7981 100644
--- a/dist/types/ArgumentException.d.ts
+++ b/dist/types/ArgumentException.d.ts	
@@ -3,7 +3,7 @@
  * Licensing: MIT
  * Based upon: https://msdn.microsoft.com/en-us/library/System.Exception%28v=vs.110%29.aspx
  */
-import SystemException from './SystemException';
+import SystemException from './SystemException.js';
 export default class ArgumentException extends SystemException {
     readonly paramName: string;
     constructor(paramName: string, message?: string, innerException?: unknown);
diff --git a/dist/types/ArgumentNullException.d.ts b/dist/types/ArgumentNullException.d.ts
index 4f1a28c7cdfffba304c843242978073b3fd323bd..7cb8eedbfd480cecc8f18ce92877440ed2e3a9f5 100644
--- a/dist/types/ArgumentNullException.d.ts
+++ b/dist/types/ArgumentNullException.d.ts	
@@ -3,7 +3,7 @@
  * Licensing: MIT
  * Based upon: https://msdn.microsoft.com/en-us/library/System.Exception%28v=vs.110%29.aspx
  */
-import ArgumentException from './ArgumentException';
+import ArgumentException from './ArgumentException.js';
 export default class ArgumentNullException extends ArgumentException {
     constructor(paramName: string, message?: string, innerException?: unknown);
     protected getName(): string;
diff --git a/dist/types/ArgumentOutOfRangeException.d.ts b/dist/types/ArgumentOutOfRangeException.d.ts
index d19caeec9a613440b82bf4287788193420f5b743..1c47b10b7edd828e1a272ee012ac6665673edd67 100644
--- a/dist/types/ArgumentOutOfRangeException.d.ts
+++ b/dist/types/ArgumentOutOfRangeException.d.ts	
@@ -3,7 +3,7 @@
  * Licensing: MIT
  * Based upon: https://msdn.microsoft.com/en-us/library/System.Exception%28v=vs.110%29.aspx
  */
-import ArgumentException from './ArgumentException';
+import ArgumentException from './ArgumentException.js';
 type Primitive = string | number | boolean;
 export default class ArgumentOutOfRangeException extends ArgumentException {
     readonly actualValue: Primitive | null | undefined;
diff --git a/dist/types/InvalidOperationException.d.ts b/dist/types/InvalidOperationException.d.ts
index 3414c8b8f94d3dbc1f909cdf81f52933d503c6bb..2d2148c45a587f6a5649d2cb589bc23a6b4b6685 100644
--- a/dist/types/InvalidOperationException.d.ts
+++ b/dist/types/InvalidOperationException.d.ts	
@@ -3,7 +3,7 @@
  * Licensing: MIT
  * Based upon: https://msdn.microsoft.com/en-us/library/System.Exception%28v=vs.110%29.aspx
  */
-import SystemException from './SystemException';
+import SystemException from './SystemException.js';
 export default class InvalidOperationException extends SystemException {
     protected getName(): string;
 }
diff --git a/dist/types/NotImplementedException.d.ts b/dist/types/NotImplementedException.d.ts
index 43a59e23f16d68725842354f3efc0b4bc3547cc5..ceeb124d4edd673fc4dc6aa7d6f8168933271ef5 100644
--- a/dist/types/NotImplementedException.d.ts
+++ b/dist/types/NotImplementedException.d.ts	
@@ -3,7 +3,7 @@
  * Licensing: MIT
  * Based upon: https://msdn.microsoft.com/en-us/library/System.Exception%28v=vs.110%29.aspx
  */
-import SystemException from './SystemException';
+import SystemException from './SystemException.js';
 export default class NotImplementedException extends SystemException {
     protected getName(): string;
 }
diff --git a/dist/types/NullReferenceException.d.ts b/dist/types/NullReferenceException.d.ts
index b2a2027d15b52db872484f683ed032ff7a9bb03c..bb22a06bba80e8d9c69161fa2d229910673b6bc7 100644
--- a/dist/types/NullReferenceException.d.ts
+++ b/dist/types/NullReferenceException.d.ts	
@@ -3,7 +3,7 @@
  * Licensing: MIT
  * Based upon: https://msdn.microsoft.com/en-us/library/System.Exception%28v=vs.110%29.aspx
  */
-import SystemException from './SystemException';
+import SystemException from './SystemException.js';
 export default class NullReferenceException extends SystemException {
     protected getName(): string;
 }
diff --git a/dist/types/SystemException.d.ts b/dist/types/SystemException.d.ts
index 3176012ffce22de9c1b3f503a4e3c4a96cf05cfe..fb608e1a1f26176381b0cfb427893c8048b27396 100644
--- a/dist/types/SystemException.d.ts
+++ b/dist/types/SystemException.d.ts	
@@ -3,7 +3,7 @@
  * Licensing: MIT
  * Based upon: https://msdn.microsoft.com/en-us/library/system.systemexception%28v=vs.110%29.aspx
  */
-import Exception from './Exception';
+import Exception from './Exception.js';
 export default class SystemException extends Exception {
     protected getName(): string;
 }
diff --git a/dist/types/index.d.ts b/dist/types/index.d.ts
index 17664d2aa233f9f17ebc1bf16a88d84f0b21f46e..7ee29ead7bde5b59497cf7a12dfe9e82e49c1905 100644
--- a/dist/types/index.d.ts
+++ b/dist/types/index.d.ts	
@@ -1,10 +1,10 @@
-import Exception from './Exception';
-import SystemException from './SystemException';
-import ArgumentException from './ArgumentException';
-import ArgumentNullException from './ArgumentNullException';
-import ArgumentOutOfRangeException from './ArgumentOutOfRangeException';
-import InvalidOperationException from './InvalidOperationException';
-import NotImplementedException from './NotImplementedException';
-import NullReferenceException from './NullReferenceException';
+import Exception from './Exception.js';
+import SystemException from './SystemException.js';
+import ArgumentException from './ArgumentException.js';
+import ArgumentNullException from './ArgumentNullException.js';
+import ArgumentOutOfRangeException from './ArgumentOutOfRangeException.js';
+import InvalidOperationException from './InvalidOperationException.js';
+import NotImplementedException from './NotImplementedException.js';
+import NullReferenceException from './NullReferenceException.js';
 export { Exception, SystemException, ArgumentException, ArgumentNullException, ArgumentOutOfRangeException, InvalidOperationException, NotImplementedException, NullReferenceException };
 export default Exception;
