diff --git a/dist/types/Collection.d.ts b/dist/types/Collection.d.ts
index 9bb4a2eda188f73504a6a71173b0b84185f50108..8965c9bce635750aae16f73fe65ecb21acb4ff63 100644
--- a/dist/types/Collection.d.ts
+++ b/dist/types/Collection.d.ts	
@@ -1,10 +1,10 @@
-﻿
+
 /*!
  * @author electricessence / https://github.com/electricessence/
  * @license MIT
  */
 
-import ReadOnlyCollection from './ReadOnlyCollection';
+import ReadOnlyCollection from './ReadOnlyCollection.js';
 
 /**
  * Interface for implementing an externally modifiable collection.
diff --git a/dist/types/CollectionBase.d.ts b/dist/types/CollectionBase.d.ts
index 79d8936952e9f7620516339af97ef2e61959ff5a..85b9909f04d14b45e8def3c557b8e8c3724bbb98 100644
--- a/dist/types/CollectionBase.d.ts
+++ b/dist/types/CollectionBase.d.ts	
@@ -3,8 +3,8 @@
  * @license MIT
  */
 import { EqualityComparison } from '@tsdotnet/compare';
-import Collection from './Collection';
-import ReadOnlyCollectionBase from './ReadOnlyCollectionBase';
+import Collection from './Collection.js';
+import ReadOnlyCollectionBase from './ReadOnlyCollectionBase.js';
 export default abstract class CollectionBase<T> extends ReadOnlyCollectionBase<T> implements Collection<T> {
     private _count;
     protected constructor(equalityComparer?: EqualityComparison<T>);
diff --git a/dist/types/IndexedCollection.d.ts b/dist/types/IndexedCollection.d.ts
index 391eefe94a8f2950b76ef3649fd01fa8ef89f6eb..40df6389bef2c0d1e9ef3bb731897d455c0eaabe 100644
--- a/dist/types/IndexedCollection.d.ts
+++ b/dist/types/IndexedCollection.d.ts	
@@ -1,11 +1,11 @@
-﻿
+
 /*!
  * @author electricessence / https://github.com/electricessence/
  * @license MIT
  */
 
-import Collection from './Collection';
-import ReadOnlyList from './ReadOnlyList';
+import Collection from './Collection.js';
+import ReadOnlyList from './ReadOnlyList.js';
 
 /**
  * Interface for implementing an indexable collection.
diff --git a/dist/types/IterableCollectionBase.d.ts b/dist/types/IterableCollectionBase.d.ts
index c6a7a1cd1fbe1af0fbbbc61694706c152e16201e..8da8d242489633b8e054553d603b9f43cf12ed03 100644
--- a/dist/types/IterableCollectionBase.d.ts
+++ b/dist/types/IterableCollectionBase.d.ts	
@@ -2,7 +2,7 @@
  * @author electricessence / https://github.com/electricessence/
  * @license MIT
  */
-import ReadOnlyIterableCollectionBase from './ReadOnlyIterableCollectionBase';
+import ReadOnlyIterableCollectionBase from './ReadOnlyIterableCollectionBase.js';
 export default abstract class IterableCollectionBase<T> extends ReadOnlyIterableCollectionBase<T> {
     protected constructor();
     private _version?;
diff --git a/dist/types/ReadOnlyCollection.d.ts b/dist/types/ReadOnlyCollection.d.ts
index 0b9ad53483e4ac49f739bf5e97830be22f4f9f4e..2a31377251a58d950bfe8009a670d0ed3e2f12d7 100644
--- a/dist/types/ReadOnlyCollection.d.ts
+++ b/dist/types/ReadOnlyCollection.d.ts	
@@ -4,7 +4,7 @@
  * @license MIT
  */
 
-import IterableCollectionBase from './IterableCollectionBase';
+import IterableCollectionBase from './IterableCollectionBase.js';
 
 /**
  * Interface for implementing a finite read-only collection.
diff --git a/dist/types/ReadOnlyCollectionBase.d.ts b/dist/types/ReadOnlyCollectionBase.d.ts
index eb25083276b8f81571c2524761de2568b2273685..79839c21bd87e077f53393cda3c3bc38fc5efc8a 100644
--- a/dist/types/ReadOnlyCollectionBase.d.ts
+++ b/dist/types/ReadOnlyCollectionBase.d.ts	
@@ -3,8 +3,8 @@
  * @license MIT
  */
 import { EqualityComparison } from '@tsdotnet/compare';
-import IterableCollectionBase from './IterableCollectionBase';
-import ReadOnlyCollection from './ReadOnlyCollection';
+import IterableCollectionBase from './IterableCollectionBase.js';
+import ReadOnlyCollection from './ReadOnlyCollection.js';
 export default abstract class ReadOnlyCollectionBase<T> extends IterableCollectionBase<T> implements ReadOnlyCollection<T> {
     protected _equalityComparer: EqualityComparison<T>;
     protected constructor(_equalityComparer?: EqualityComparison<T>);
diff --git a/dist/types/ReadOnlyIterableCollectionBase.d.ts b/dist/types/ReadOnlyIterableCollectionBase.d.ts
index 2bf4790a7c8268b432eefd4f878a25aaed119c0a..4f08e586b998bda08023962a87b3a411b86913c5 100644
--- a/dist/types/ReadOnlyIterableCollectionBase.d.ts
+++ b/dist/types/ReadOnlyIterableCollectionBase.d.ts	
@@ -3,7 +3,7 @@
  * @license MIT
  */
 import { ArrayLikeWritable, PredicateWithIndex, SelectorWithIndex } from '@tsdotnet/common-interfaces';
-import IterableBase from './IterableBase';
+import IterableBase from './IterableBase.js';
 export default abstract class ReadOnlyIterableCollectionBase<T> extends IterableBase<T> {
     protected constructor();
     filter(predicate: PredicateWithIndex<T>): ExtendedIterable<T>;
diff --git a/dist/types/ReadOnlyList.d.ts b/dist/types/ReadOnlyList.d.ts
index ee5e64e21b11fa6a2c6f5943e8017b6c30389449..d7eed3e5fcb55d63078f25de52239261c8c68cb9 100644
--- a/dist/types/ReadOnlyList.d.ts
+++ b/dist/types/ReadOnlyList.d.ts	
@@ -4,7 +4,7 @@
  * @license MIT
  */
 
-import ReadOnlyCollection from './ReadOnlyCollection';
+import ReadOnlyCollection from './ReadOnlyCollection.js';
 
 export default interface ReadOnlyList<T>
 	extends ReadOnlyCollection<T>
diff --git a/dist/types/collection-base.d.ts b/dist/types/collection-base.d.ts
index 03465c15c719ff4b68120962979924fc6a9d2e8c..c762a01a9b6a5155c9e90d1e433acbd84fd311dc 100644
--- a/dist/types/collection-base.d.ts
+++ b/dist/types/collection-base.d.ts	
@@ -1,13 +1,13 @@
-import arrayLikeIterable from './arrayLikeIterable';
-import type Collection from './Collection';
-import CollectionBase from './CollectionBase';
-import copyIterableTo from './copyIterableTo';
-import type IndexedCollection from './IndexedCollection';
-import indexIterable from './indexIterable';
-import IterableBase from './IterableBase';
-import IterableCollectionBase from './IterableCollectionBase';
-import type ReadOnlyCollection from './ReadOnlyCollection';
-import ReadOnlyCollectionBase from './ReadOnlyCollectionBase';
-import ReadOnlyIterableCollectionBase, { ExtendedIterable } from './ReadOnlyIterableCollectionBase';
-import type ReadOnlyList from './ReadOnlyList';
+import arrayLikeIterable from './arrayLikeIterable.js';
+import type Collection from './Collection.js';
+import CollectionBase from './CollectionBase.js';
+import copyIterableTo from './copyIterableTo.js';
+import type IndexedCollection from './IndexedCollection.js';
+import indexIterable from './indexIterable.js';
+import IterableBase from './IterableBase.js';
+import IterableCollectionBase from './IterableCollectionBase.js';
+import type ReadOnlyCollection from './ReadOnlyCollection.js';
+import ReadOnlyCollectionBase from './ReadOnlyCollectionBase.js';
+import ReadOnlyIterableCollectionBase, { ExtendedIterable } from './ReadOnlyIterableCollectionBase.js';
+import type ReadOnlyList from './ReadOnlyList.js';
 export { copyIterableTo, arrayLikeIterable, indexIterable, ReadOnlyCollection, Collection, ReadOnlyList, IndexedCollection, IterableBase, IterableCollectionBase, ExtendedIterable, ReadOnlyCollectionBase, ReadOnlyIterableCollectionBase, CollectionBase };
